atom/common/api/atom_bindings.cc

```diff
@@ -239,28 +239,19 @@ v8::Local<v8::Promise> AtomBindings::GetProcessMemoryInfo(
     return promise->GetHandle();
   }
 
-  v8::Global<v8::Context> context(isolate, isolate->GetCurrentContext());
   memory_instrumentation::MemoryInstrumentation::GetInstance()
-      ->RequestGlobalDumpForPid(base::GetCurrentProcId(),
-                                std::vector<std::string>(),
-                                base::Bind(&AtomBindings::DidReceiveMemoryDump,
-                                           std::move(context), promise));
+      ->RequestGlobalDumpForPid(
+          base::GetCurrentProcId(), std::vector<std::string>(),
+          base::Bind(&AtomBindings::DidReceiveMemoryDump, promise));
   return promise->GetHandle();
 }
 
 // static
 void AtomBindings::DidReceiveMemoryDump(
-    const v8::Global<v8::Context>& context,
     scoped_refptr<util::Promise> promise,
     bool success,
     std::unique_ptr<memory_instrumentation::GlobalMemoryDump> global_dump) {
   v8::Isolate* isolate = promise->isolate();
-  mate::Locker locker(isolate);
```

Comment:
Although `util::promise` enters the right scope for its methods, the `mate::dictionary` methods below still require the scope entry.